flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2570,12 +2570,14 @@ def multistep_change_log(parent, children):
     """
     def verify(self, logger):
         [parent_action] = LoggedAction.of_type(logger.messages, parent)
-        children_actions = [
-            LoggedAction.of_type(logger.messages, child_action)[0]
-            for child_action
-            in children
-        ]
-        self.assertEqual(children_actions, parent_action.children)
+        children_actions = {
+            child_action.action_type:
+            LoggedAction.of_type(logger.messages, child_action)
+            for child_action in children
+        }
+        for child in parent_action.children:
+            action_type = child.startMessage["action_type"]
+            self.assertIn(child, children_actions.get(action_type, []))
```

Comment:
It does still verify order. The issue is just that because the CreateVolume now logs sub-actions you get multiple mounts logged in the test. So you assert requested children happened in correct order but don't blow if child occurred earlier in test, when it's logs for pre-test-assertion setup.